pandas/tests/frame/indexing/test_mask.py

```diff
@@ -150,3 +150,12 @@ def test_mask_inplace_no_other():
     df.mask(cond, inplace=True)
     expected = DataFrame({"a": [np.nan, 2], "b": ["x", np.nan]})
     tm.assert_frame_equal(df, expected)
+
+
+def test_mask_with_NA():
+    df = DataFrame({"A": [0, 1, 2]})
+    cond = Series([True, False, pd.NA], dtype=pd.BooleanDtype())
```

Comment:
To pass the tests, I think `dtype=BooleanDtype()` should be used here, and from pandas import `BooleanDtype` at the top of the file, just like other dtypes.